contracts/ownable/src/test/simulators/OwnableSimulator.ts

```diff
@@ -0,0 +1,229 @@
+ */
+export class OwnableSimulator
+  implements IContractSimulator<OwnablePrivateState, Ledger>
+{
+  /** @description The underlying contract instance managing contract logic. */
+  readonly contract: MockOwnable<OwnablePrivateState>;
+
+  /** @description The deployed address of the contract. */
+  readonly contractAddress: string;
+
+  /** @description The current circuit context, updated by contract operations. */
+  circuitContext: CircuitContext<OwnablePrivateState>;
+
+  /**
+   * @description Initializes the mock contract.
+   */
+  constructor(
+    initialOwner: Either<ZswapCoinPublicKey, ContractAddress>,
+    isInit: boolean,
+  ) {
+    this.contract = new MockOwnable<OwnablePrivateState>(OwnableWitnesses);
+    const {
+      currentPrivateState,
+      currentContractState,
+      currentZswapLocalState,
+    } = this.contract.initialState(
+      constructorContext({}, '0'.repeat(64)),
+      initialOwner,
+      isInit,
+    );
+    this.circuitContext = {
+      currentPrivateState,
+      currentZswapLocalState,
+      originalState: currentContractState,
+      transactionContext: new QueryContext(
+        currentContractState.data,
+        sampleContractAddress(),
+      ),
+    };
+    this.contractAddress = this.circuitContext.transactionContext.address;
+  }
+
+  /**
+   * @description Retrieves the current public ledger state of the contract.
+   * @returns The ledger state as defined by the contract.
+   */
+  public getCurrentPublicState(): Ledger {
+    return ledger(this.circuitContext.transactionContext.state);
+  }
+
+  /**
+   * @description Retrieves the current private state of the contract.
+   * @returns The private state of type OwnablePrivateState.
+   */
+  public getCurrentPrivateState(): OwnablePrivateState {
+    return this.circuitContext.currentPrivateState;
+  }
+
+  /**
+   * @description Retrieves the current contract state.
+   * @returns The contract state object.
+   */
+  public getCurrentContractState(): ContractState {
+    return this.circuitContext.originalState;
+  }
+
+  /**
+   * @description Returns the current contract owner.
+   * @returns The contract owner.
+   */
+  public owner(): Either<ZswapCoinPublicKey, ContractAddress> {
+    return this.contract.impureCircuits.owner(this.circuitContext).result;
+  }
+
+  /**
+   * @description Transfers ownership of the contract to `newOwner`.
+   * @param newOwner - The new owner.
+   * @param sender - Optional. Sets the caller context if provided.
+   */
+  public transferOwnership(
+    newOwner: Either<ZswapCoinPublicKey, ContractAddress>,
+    sender?: CoinPublicKey,
+  ) {
+    const res = this.contract.impureCircuits.transferOwnership(
+      {
+        ...this.circuitContext,
+        currentZswapLocalState: sender
+          ? emptyZswapLocalState(sender)
+          : this.circuitContext.currentZswapLocalState,
+      },
+      newOwner,
+    );
+
+    this.circuitContext = res.context;
+    return res.result;
```

Comment:
We can remove this return along with all others (other than `owner()`) since the circuits don't return anything.
```suggestion
```